pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/kit/ClinicalKitDto.java

```diff
@@ -101,30 +102,33 @@ public void setGender(String genderString) {
                 ElasticSearchUtil.getParticipantESDataByParticipantId(ddpInstance.getParticipantIndexES(), ddpParticipantId);
         if (maybeParticipantESDataByParticipantId.isEmpty()) {
-            throw new RuntimeException("Participant ES Data is not found for " + ddpParticipantId);
+            throw new DsmInternalError("Participant ES Data is not found for " + ddpParticipantId);
         }
-
+        ElasticSearchParticipantDto elasticSearchParticipantDto = maybeParticipantESDataByParticipantId.get();
         try {
-            this.setDateOfBirth(maybeParticipantESDataByParticipantId.get().getDsm().map(Dsm::getDateOfBirth).orElse(""));
-            this.setFirstName(maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getFirstName).orElse(""));
-            this.setLastName(maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getLastName).orElse(""));
-            this.setGender(getParticipantGender(maybeParticipantESDataByParticipantId.get(), ddpInstance.getName()));
-            String shortId = maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getHruid).orElse("");
+            this.setDateOfBirth(elasticSearchParticipantDto.getDsm().map(Dsm::getDateOfBirth).orElse(""));
+            this.setFirstName(elasticSearchParticipantDto.getProfile().map(Profile::getFirstName).orElse(""));
+            this.setLastName(elasticSearchParticipantDto.getProfile().map(Profile::getLastName).orElse(""));
+            this.setGender(getParticipantGender(elasticSearchParticipantDto, ddpInstance.getName()));
+            String shortId = elasticSearchParticipantDto.getProfile().map(Profile::getHruid).orElse("");
             String collaboratorParticipantId =
                     KitRequestShipping.getCollaboratorParticipantId(ddpInstance.getBaseUrl(), ddpInstance.getDdpInstanceId(),
-                            ddpInstance.isMigratedDDP(),
-                            ddpInstance.getCollaboratorIdPrefix(), ddpParticipantId, shortId, null);
+                            ddpInstance.isMigratedDDP(), ddpInstance.getCollaboratorIdPrefix(), ddpParticipantId, shortId, null);
             this.setCollaboratorParticipantId(collaboratorParticipantId);
         } catch (Exception e) {
-            throw new RuntimeException("Participant doesn't exist / is not valid for kit " + e.getMessage());
+            throw new DsmInternalError(String.format("Error getting participant data for %s", ddpParticipantId), e);
         }
     }
 
     private String getParticipantGender(ElasticSearchParticipantDto participantByShortId, String realm) {
         // if gender is set on tissue page use that
+        if(StringUtils.isBlank(participantByShortId.getParticipantId())){
+            throw new DsmInternalError(String.format("The participant %s is missing participant id", participantByShortId.getProfile().get().getGuid()));
+        }
```

Comment:
We have not established that the ptp has a profile at this point, so this may fail. But we do have a ddpParticipantId that we can pass into this method.
(This is also a good place to assign the return value of `participantByShortId.getParticipantId()` to improve readability.)